mkosi/util.py

```diff
@@ -321,3 +267,14 @@ def flock(path: Path) -> Iterator[int]:
         yield fd
     finally:
         os.close(fd)
+
+
+@contextlib.contextmanager
+def withenv(env: Mapping[str, Any]) -> Iterator[None]:
```

Comment:
Bikeshed: `with withenv` sounds a bit weird, how about `tempenv`?

Reply:
Renamed to `scopedenv()`